tests/samples_test.py

```diff
@@ -0,0 +1,416 @@
+    assert f.frequency == -5
+    assert isinstance(f.frequency, int)
+
+
+@pytest.mark.parametrize(
+    "nested_samples",
+    [
+        smolyay.samples.NestedClenshawCurtisPointSet,
+        smolyay.samples.SlowNestedClenshawCurtisPointSet,
+        smolyay.samples.NestedTrigonometricPointSet,
+    ],
+    ids=[
+        "NestedClenshawCurtis",
+        "SlowNestedClenshawCurtis",
+        "NestedTrigonometric",
+    ],
+)
+def test_initialize_nested(nested_samples):
+    """test default properties"""
+    f = nested_samples([-10, 10], 7)
+    assert numpy.array_equal(f.domain, [-10, 10])
+    assert f.num_level == 7
+    assert isinstance(f.num_level, int)
+    f.num_level = float(5)
+    assert f.num_level == 5
+    assert isinstance(f.num_level, int)
+
+
+@pytest.mark.parametrize(
+    "samples,set_args",
+    [
+        (smolyay.samples.ClenshawCurtisPointSet, {"degree": 7}),
+        (smolyay.samples.TrigonometricPointSet, {"frequency": 7}),
+        (smolyay.samples.NestedClenshawCurtisPointSet, {"num_level": 7}),
+        (smolyay.samples.SlowNestedClenshawCurtisPointSet, {"num_level": 7}),
+        (smolyay.samples.NestedTrigonometricPointSet, {"num_level": 7}),
+    ],
+    ids=[
+        "ClenshawCurtis",
+        "Trigonometric",
+        "NestedClenshawCurtis",
+        "SlowNestedClenshawCurtis",
+        "NestedTrigonometric",
+    ],
+)
+def test_domain_error(samples, set_args):
+    """test error given invalid domain and that reversed domains swap"""
+    # reverse domain
+    f = samples([10, -10], **set_args)
+    assert numpy.array_equal(f.domain, [-10, 10])
+    f = samples([-10, 10], **set_args)
+    f.domain = [5, -10]
+    assert numpy.array_equal(f.domain, [-10, 5])
+    # invalid domain
+    with pytest.raises(TypeError):
+        samples([-10, 10, 20], **set_args)
+    with pytest.raises(TypeError):
+        samples([[-10, 10], [-10, 10]], **set_args)
+    with pytest.raises(ValueError):
+        samples([10, 10], **set_args)
+    with pytest.raises(TypeError):
+        f = samples([-10, 10], **set_args)
+        f.domain = [[-10, 10], [-10, 10]]
+    with pytest.raises(TypeError):
+        f = samples([-10, 10], **set_args)
+        f.domain = [-10, 10, 20]
+    with pytest.raises(ValueError):
+        f = samples([-10, 10], **set_args)
+        f.domain = [10, 10]
+
+
+@pytest.mark.parametrize(
+    "nested_samples",
+    [
+        smolyay.samples.NestedClenshawCurtisPointSet,
+        smolyay.samples.SlowNestedClenshawCurtisPointSet,
+        smolyay.samples.NestedTrigonometricPointSet,
+    ],
+    ids=[
+        "NestedClenshawCurtis",
+        "SlowNestedClenshawCurtis",
+        "NestedTrigonometric",
+    ],
+)
+def test_num_level_error(nested_samples):
+    """test error given invalid num_level"""
+    with pytest.raises(ValueError):
+        f = nested_samples([-10, 10], 0)
+        f.domain = [[-10, 10], [-10, 10]]
+    with pytest.raises(ValueError):
+        f = nested_samples([-10, 10], 2)
```

Comment:
This should go outside the test, so you make sure it constructs correctly and then only errors on set.
```suggestion
    f = nested_samples([-10, 10], 2)
    with pytest.raises(ValueError):
```